saleor/graphql/webhook/mutations.py

```diff
@@ -144,14 +131,24 @@ class Meta:
         error_type_field = "webhook_errors"
 
     @classmethod
-    def clean_input(cls, info, instance, data, input_cls=None):
-        cleaned_input = super().clean_input(info, instance, data, input_cls)
-        # This should be removed after we drop service_account
-        service_account = cleaned_input.pop("service_account", None)
-        app = cleaned_input.get("app", service_account)
-        if app:
-            cleaned_input["app"] = app
-        return cleaned_input
+    def clean_input(cls, info, instance, data):
+        cleaned_data = super().clean_input(info, instance, data)
+        app = cleaned_data.get("app")
+
+        if not instance.app_id and not app:
+            raise ValidationError("Missing token or app", code=WebhookErrorCode.INVALID)
+
+        if instance.app_id:
+            # Let's skip app id in case when context has
+            # app instance
+            app = instance.app
+            cleaned_data.pop("app", None)
+
+        if not app or not app.is_active:
+            raise ValidationError(
+                "App doesn't exist or is disabled", code=WebhookErrorCode.NOT_FOUND,
+            )
```

Comment:
Could possibly be divided into two error cases (`WebhookErrorCode.INVALID` / `WebhookErrorCode.NOT_FOUND`)